pkg/asset/agent/image/agentimage.go

```diff
@@ -30,56 +34,111 @@ func (a *AgentImage) Dependencies() []asset.Asset {
 	return []asset.Asset{
 		&Ignition{},
 		&BaseIso{},
+		&manifests.AgentManifests{},
+		&mirror.RegistriesConf{},
 	}
 }
 
 // Generate generates the image file for to ISO asset.
 func (a *AgentImage) Generate(dependencies asset.Parents) error {
 	ignition := &Ignition{}
-	dependencies.Get(ignition)
-
 	baseImage := &BaseIso{}
-	dependencies.Get(baseImage)
+	agentManifests := &manifests.AgentManifests{}
+	registriesConf := &mirror.RegistriesConf{}
+
+	dependencies.Get(ignition, baseImage, agentManifests, registriesConf)
 
 	ignitionByte, err := json.Marshal(ignition.Config)
 	if err != nil {
 		return err
 	}
 
-	ignitionContent := &isoeditor.IgnitionContent{Config: ignitionByte}
-	custom, err := isoeditor.NewRHCOSStreamReader(baseImage.File.Filename, ignitionContent, nil)
+	agentTuiFile, err := a.fetchAgentTuiBinary(agentManifests.ClusterImageSet.Spec.ReleaseImage, agentManifests.GetPullSecretData(), registriesConf.MirrorConfig)
+	if err != nil {
+		return err
+	}
+
+	err = a.prepareAgentISO(baseImage.File.Filename, ignitionByte, agentTuiFile)
 	if err != nil {
 		return err
 	}
 
-	a.imageReader = custom
 	a.cpuArch = ignition.CPUArch
 	a.rendezvousIP = ignition.RendezvousIP
 
 	return nil
 }
 
+func (a *AgentImage) fetchAgentTuiBinary(releaseImage string, pullSecret string, mirrorConfig []mirror.RegistriesConfig) (string, error) {
+	r := NewRelease(&executer.CommonExecuter{},
+		Config{MaxTries: OcDefaultTries, RetryDelay: OcDefaultRetryDelay},
+		releaseImage, pullSecret, mirrorConfig)
+
+	// TODO: This is just a temporary placeholder to test the entire workflow.
+	// As soon as https://github.com/openshift/assisted-installer-agent/pull/482 will land, then
+	// the following line should be fixed to extract the agent-tui binary
+	filename, err := r.ExtractFile("agent-installer-node-agent", "/usr/bin/agent")
```

Comment:
```suggestion
	filename, err := r.ExtractFile("agent-installer-node-agent", "/usr/local/bin/agent-tui")
```
From my understanding /usr/bin is usually used by packages installed by the OS's package manager.

/usr/local/bin can be used for user installed apps. The existing assisted-service agent binary is written to /usr/local/bin/agent. So I suggest we use /usr/local/bin/agent-tui.

Reply:
This is the source location (inside the `agent-installer-node-agent` image) from where the [binary](https://github.com/openshift/assisted-installer-agent/blob/9ae865b4ed549c5ed5ba3b8978a780fae2246bca/Dockerfile.ocp#L12) is extracted.
As mentioned in the comment, this is just a mere placeholder to verify that the entire workflow works as expected, and it will have to be changed as soon as the agent-tui binary will be available.

Reply:
Ah, my bad. Looks fine then.